stan/math/prim/err/check_greater.hpp

```diff
@@ -2,78 +2,253 @@
+ * @tparam T_y A standard vector
+ * @tparam T_low A standard vector
  * @param function Function name (for error messages)
  * @param name Variable name (for error messages)
  * @param y Variable to check
  * @param low Lower bound
- * @throw <code>domain_error</code> if y is not greater than low or
+ * @throw <code>domain_error</code> if y is not greater or equal to low or
  *   if any element of y or low is NaN.
  */
-template <typename T_y, typename T_low>
+template <typename T_y, typename T_low,
+          require_all_std_vector_t<T_y, T_low>* = nullptr>
 inline void check_greater(const char* function, const char* name, const T_y& y,
                           const T_low& low) {
-  internal::greater<T_y, T_low, is_vector_like<T_y>::value>::check(
-      function, name, y, low);
+  check_matching_sizes(function, name, y, "lower", low);
+  for (size_t i = 0; i < y.size(); ++i) {
+    check_greater(function, internal::make_iter_name(name, i).c_str(), y[i],
+                  low[i]);
+  }
 }
+
+/**
+ * Check if each element of <code>y</code> is strictly greater or equal than
+ * <code>low</code>. This function is vectorized and will check each element of
+ * <code>y</code> against each element of <code>low</code>.
+ * @tparam T_y A standard vector
+ * @tparam T_low A scalar or the same type as the underlying type in `T_y`
+ * @param function Function name (for error messages)
+ * @param name Variable name (for error messages)
+ * @param y Variable to check
+ * @param low Lower bound
+ * @throw <code>domain_error</code> if y is not greater or equal to low or
+ *   if any element of y or low is NaN.
+ */
+template <typename T_y, typename T_low, require_std_vector_t<T_y>* = nullptr,
+          require_not_std_vector_t<T_low>* = nullptr>
+inline void check_greater(const char* function, const char* name, const T_y& y,
+                          const T_low& low) {
+  for (size_t i = 0; i < y.size(); ++i) {
+    check_greater(function, internal::make_iter_name(name, i).c_str(), y[i],
+                  low);
+  }
+}
+
+/**
+ * Check if each element of <code>y</code> is strictly greater or equal than
+ * <code>low</code>. This function is vectorized and will check each element of
+ * <code>y</code> against each element of <code>low</code>.
+ * @tparam T_y A scalar or the same type as the inner type of `T_low`
+ * @tparam T_low A standard vector
+ * @param function Function name (for error messages)
+ * @param name Variable name (for error messages)
+ * @param y Variable to check
+ * @param low Lower bound
+ * @throw <code>domain_error</code> if y is not greater or equal to low or
+ *   if any element of y or low is NaN.
+ */
+template <typename T_y, typename T_low,
+          require_not_std_vector_t<T_y>* = nullptr,
+          require_std_vector_t<T_low>* = nullptr>
+inline void check_greater(const char* function, const char* name, const T_y& y,
+                          const T_low& low) {
+  for (size_t i = 0; i < low.size(); ++i) {
```

Comment:
This one is iterating over `low` and has has a scalar y. Since we only have the `name` of `y` we can't use `throw_domain_vec` since we don't want to attach an index to a scalar when the index is for the `low`